src/utils.ts

```diff
@@ -1,32 +1,56 @@
 import * as cheerio from "cheerio";
-import { Err, Ok } from "./graduate-types/common";
-import type { Result } from "./graduate-types/common";
-import undici from "undici";
+import { log } from "@clack/prompts";
+import { ResultType, type Matcher, type Result, Err, Ok } from "@/types";
 
-export const loadHTML = async (url: string): Promise<CheerioStatic> => {
-  return cheerio.load(await wrappedGetRequest(url));
-};
+export async function retryFetchHTML(
+  url: URL,
+  numRetries: number = 5,
+  sleepTime: number = 500,
+): Promise<Result<CheerioStatic, string>> {
+  for (let i = 0; i < numRetries - 1; i++) {
+    const res = await fetchHTML(url);
+
+    if (res.type === ResultType.Ok) {
+      return res;
+    }
+
+    // Network fails like 404 etc
+    if (res.type === ResultType.Err && !res.err.includes("Fetch failed")) {
+      return res;
+    }
+
+    setTimeout(() => {}, sleepTime);
+  }
 
-export const loadHtmlWithUrl = async (
+  return await fetchHTML(url);
+}
+
+export async function fetchHTML(
   url: URL,
-): Promise<{ url: URL; result: Result<CheerioStatic, unknown> }> => {
-  let result: Result<CheerioStatic, unknown>;
+): Promise<Result<CheerioStatic, string>> {
   try {
-    const html = cheerio.load(await wrappedGetRequest(url.href));
-    result = Ok(html);
-  } catch (error) {
-    result = Err(error);
-  }
-  return { url, result };
-};
+    const res = await fetch(url, {
```

Comment:
fetch over unidici 👀 Is it because it works more consistently?

Reply:
not really, `fetch` uses `undici` under the hood in `node`. Alex and Cindy used `undici` probably because they were using `node 16` and `fetch` API wasn't available until `node 18`